ion-schema/src/isl/isl_type.rs

```diff
@@ -55,11 +59,22 @@ impl IslType {
 pub struct IslTypeImpl {
     name: Option<String>,
     constraints: Vec<IslConstraint>,
+    // Represents the ISL type struct in string format for anonymous type definition
+    // For named type definition, this will be `None`
+    pub(crate) isl_type_struct: Option<String>,
```

Comment:
What's the rationale for making this optional instead of just using the ISL regardless of whether it's a named or anonymous type?

Reply:
Thats right, I can do it for all type definitions irrespective of named or anonymous but what I missed in the document comment is this field is set to `None` also when the ISL type definition is generated programmatically using `new` method?